go/libraries/doltcore/env/actions/checkout.go

```diff
@@ -49,14 +49,21 @@ func MoveTablesFromHeadToWorking(ctx context.Context, roots doltdb.Roots, tbls [
 	var unknownTbls []string
 	for _, tblName := range tbls {
 		tbl, ok, err := roots.Staged.GetTable(ctx, tblName)
-
+		if err != nil {
+			return doltdb.Roots{}, err
+		}
+		fkc, err := roots.Staged.GetForeignKeyCollection(ctx)
 		if err != nil {
 			return doltdb.Roots{}, err
 		}
 
 		if !ok {
 			tbl, ok, err = roots.Head.GetTable(ctx, tblName)
+			if err != nil {
+				return doltdb.Roots{}, err
+			}
 
+			fkc, err = roots.Staged.GetForeignKeyCollection(ctx)
```

Comment:
Can you reuse the `fkc` data from line 55 and avoid calling `GetForeignKeyCollection` again?